ocaml/xapi/create_misc.ml

```diff
@@ -337,6 +337,15 @@ and create_domain_zero_console_record_with_protocol ~__context ~domain_zero_ref
   let address =
     Db.Host.get_address ~__context ~self:(Helpers.get_localhost ~__context)
   in
+  let address =
+    match Xapi_stdext_unix.Unixext.domain_of_addr address with
+    | Some x when x = Unix.PF_INET ->
+      address
+    | Some x when x = Unix.PF_INET6 ->
+      Http.Url.maybe_wrap_IPv6_literal address
+    | _ ->
+      ""
```

Comment:
it may be better to keep the old behaviour, and return `address` when `x` is a unix socket (though it shouldn't make a difference as far as I can tell). It would also simplify the match slightly:

```
let address =
  match domain_of_addr address with
  | Some x when x = Unix.PF_INET6 ->
    Http.Url.maybe_wrap_IPv6_literal address
  | _ -> address
```